activerecord/lib/active_record/migration.rb

```diff
@@ -148,8 +148,14 @@ def initialize(message = nil)
         super("Migrations are pending. To resolve this issue, run:\n\n        bin/rails db:migrate RAILS_ENV=#{::Rails.env}")
       elsif !message
         super("Migrations are pending. To resolve this issue, run:\n\n        bin/rails db:migrate")
-      else
-        super
+      end
+
+      pending_migrations = ActiveRecord::Base.connection.migration_context.open.pending_migrations
+
+      @details = "You have #{pending_migrations.size} pending #{pending_migrations.size > 1 ? 'migrations:' : 'migration:'}\n\n"
```

Comment:
What do you think about including this information in the error message instead? This way, the pending migrations are shown in the logs as well as the UI.

Reply:
Sure -- I can add the detail to the message attribute, the message attribute is displayed in the log/console.

The reason I ended up splitting out into it's own attribute so we can have a clear title at the top, that is not formatted and a second attribute that adds a more detail which is formatted (it adds new lines automatically) i.e. `<%= simple_format h(@exception.details) %>`

Reply:
NOTE: once https://github.com/rails/rails/pull/39723 is merged, I will rebase this PR and use the `message` attribute instead.